src/zkevm_specs/evm_circuit/execution/create.py

```diff
@@ -130,71 +81,181 @@ def create(instruction: Instruction):
         all_but_one_64th_gas,
     )
 
-    # copy init_code from memory to bytecode
-    copy_rwc_inc, _ = instruction.copy_lookup(
-        instruction.curr.call_id,  # src_id
-        CopyDataTypeTag.Memory,  # src_type
-        instruction.next.code_hash,  # dst_id
-        CopyDataTypeTag.Bytecode,  # dst_type
-        offset,  # src_addr
-        offset + size,  # src_addr_boundary
-        FQ(0),  # dst_addr
-        size,  # length
-        instruction.curr.rw_counter + instruction.rw_counter_offset,
+    ### Do stack depth, nonce and balance pre-check
+    # ErrDepth constraint
+    is_error_depth, _ = instruction.compare(FQ(CALL_CREATE_DEPTH), depth, N_BYTES_STACK)
+    # ErrNonceUintOverflow constraint
+    is_nonce_overflow, _ = instruction.compare(nonce, nonce_prev, N_BYTES_U64)
+    # ErrInsufficientBalance constraint
+    is_insufficient_balance, _ = instruction.compare_word(Word(balance_prev.expr().n), value_word)
+
+    # pass the pre-check if none of above errors happen
+    is_precheck_ok = (
+        is_error_depth == FQ(0) and is_nonce_overflow == FQ(0) and is_insufficient_balance == FQ(0)
```

Comment:
To have deterministic circuit trace, we also need to constrain the order of these cases like https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/2d2bfc6ccf179ade1a8d063f9586b93e5283a557/zkevm-circuits/src/evm_circuit/execution/create.rs#L182-L199

Reply:
Just discussed offline and we find the cirucit should be updated to allow these exception to happen at the same time, so no need this check.

Reply:
After discussed with Han, it's possible to have nonce overflow, insufficient balance and depth too deep in the same time. So, we don't need to constraint the relationship among these three errors.